src/Microsoft.XmlSerializer.Generator/src/Microsoft/XmlSerializer/Generator/XmlSerializationWriter.cs

```diff
@@ -2080,6 +2080,14 @@ private void WriteValue(object value)
                     {
                         Writer.Write(((int)value).ToString(null, NumberFormatInfo.InvariantInfo));
                     }
+                    else if (type == typeof(TimeSpan))
```

Comment:
I saw the netfx is actually using `switch/case` instead of `if/else`
https://referencesource.microsoft.com/#System.Xml/System/Xml/Serialization/XmlSerializationWriter.cs,3049